query-engine/js-connectors/src/proxy.rs

```diff
@@ -289,16 +294,143 @@ fn js_planetscale_value_to_quaint(json_value: serde_json::Value, column_type: Co
     }
 }
 
-impl From<JSResultSet> for QuaintResultSet {
-    fn from(mut js_result_set: JSResultSet) -> Self {
+fn js_neon_value_to_quaint(json_value: serde_json::Value, column_type: ColumnType) -> QuaintValue<'static> {
+    //  Note for the future: it may be worth revisiting how much bloat so many panics with different static
+    // strings add to the compiled artefact, and in case we should come up with a restricted set of panic
+    // messages, or even find a way of removing them altogether.
```

Comment:
Why are we duplicating the `js_*_to_quaint` functions here? Isn't `ColumnType` already an abstracted version of each individual connectors' column types? If so, why do we need different logic to coerce values from each individual connectors?

Reply:
> Isn't ColumnType already an abstracted version of each individual connectors' column types?

it should be and there is an issue open for it. More context on https://github.com/prisma/prisma-engines/pull/4097#discussion_r1280539950

Reply:
> Isn't ColumnType already an abstracted version of each individual connectors' column types?

No, it's not. However, your comment seems very much in line with https://github.com/prisma/prisma-engines/pull/4097#discussion_r1280539950, and I totally agree with that strategy. Ideally, we'd first showcase the integration of Node Drivers with Prisma Client first, for the sake of "rapid iterations of development". Would you agree?

Reply:
<img width="1586" alt="image" src="https://github.com/prisma/prisma-engines/assets/25233379/960759b0-1880-42b3-b18f-94fc26e74868">

Ah, I had missed the differences between the two functions. I see them now.

So, IIUC, drivers will ensure data types are normalized so that we have a single way to coerce values rust side?

I won't bother you to change this piece of code if it's about to disappear but if you're interested, you could dry out these functions anyway:

```rust
fn planetscale_coerce(json_value: serde_json::Value, column_type: ColumnType) -> QuaintValue {
  match column_type {
    ColumnType::SpecificType => /*handle specific type*/,
    typ => base_coerce(json_value, typ)
  }
}
fn planetscale_coerce(json_value: serde_json::Value, column_type: ColumnType) -> QuaintValue {
  match column_type {
    ColumnType::SpecificType => /*handle specific type*/,
    typ => base_coerce(json_value, typ)
  }
}
fn base_coerce(json_value: serde_json::Value, column_type: ColumnType) -> QuaintValue {
  match column_type {
    // handle all cases
  }
}
```

Reply:
Implemented this dry approach (we would've done this in a future PR anyway).
Function names are slightly different to preserve what they used to be, you can CMD+F for `js_base_value_to_quaint`, `js_neon_value_to_quaint`, `js_planetscale_value_to_quaint`.